src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotator.java

```diff
@@ -196,6 +197,14 @@ public void onTraversalStart() {
         sequenceDictionary = getBestAvailableSequenceDictionary();
         variantSamples = new IndexedSampleList(samples);
 
+        // Check that the reads have compatible samples to the variants:
+        if (hasReads()) {
+            final Set<String> readsSamples = getHeaderForReads().getReadGroups().stream().map(rg -> rg.getSample()).collect(Collectors.toSet());
+            readsSamples.forEach(readSample -> {
+                if (!samples.contains(readSample))
+                    throw new UserException(String.format("Reads sample '%s' from readgroups tags does not match any sample in the variant genotypes", readSample));
```

Comment:
This checks whether the samples in the bam are present in the vcf. But the original bug report in https://github.com/broadinstitute/gatk/issues/6915 was (according to @gbrandt6 ) triggered by the reverse case where samples in the vcf don't exist in the bam.

Reply:
Upon further investigation, @jamesemery and I confirmed that this patch is correct, and the original bug report was backwards. The NPE is triggered when there are samples in the bam not present in the vcf.

Reply:
Thanks for looking into it @droazen @jamesemery I checked again and it looks like the user said there were no samples in common.

Reply:
If you look at the version that user was running the NPE is actually occurring on this line:   
`returnMap.get(ReadUtils.getSampleName(read, header)).add(read); `

Which means that the problem is caused when the returnMap (which gets generated right above from the VCF samples list) doesn't contain a sample that is present in one of the reads at that site. It sounded like the user had accentually matched up the wrong bam with the wrong VCF and thats why they got the crash. Technically what they said is accurate but not exactly what caused the bug, it was the reverse of what they reported as the bug. Given that multisample bams exist perhaps there is something more robust that can be done to fix the bug if this ever comes up again.